Sources/MessagingInApp/Views/InAppMessageView.swift

```diff
@@ -34,21 +34,33 @@ public class InAppMessageView: UIView {
     // Can set in the constructor or can set later (like if you use Storyboards)
     public var elementId: String? {
         didSet {
-            checkIfMessageAvailableToDisplay()
+            refreshView()
         }
     }
 
+    // Inline messages that have already been shown by this View instance.
+    // This is used to prevent showing the same message multiple times when the close button is pressed.
+    //
+    // When persistent vs non-persistent messages and metrics features are implemented in the SDK, this array may be
+    // replaced with a global list of shown messages.
+    var previouslyShownMessages: [Message] = []
```

Comment:
how about we use [OrderedSet](https://github.com/apple/swift-collections/blob/main/Documentation/OrderedSet.md) to manage duplicates?